redash/query_runner/big_query.py

```diff
@@ -344,4 +350,64 @@ def dry_run(self, query):
                 }
             }
 
+    @property
+    def supports_preview(self):
+        return True
+
+    def preview(self, full_table_name):
+        bigquery_service = self._get_bigquery_service()
+        project_id = self._get_project_id()
+
+        ds, _, table_name = full_table_name.partition(".")
+
+        try:
+            data = bigquery_service.tabledata().list(
+                projectId=project_id,
+                datasetId=ds,
+                tableId=table_name,
+                maxResults=100
+            ).execute()
+        except HttpError as he:
+            content = json.loads(he.content)
+            return {
+                "error": {
+                    "code": content["error"]["code"],
+                    "status": content["error"]["status"],
+                    "message": content["error"]["message"]
+                }
+            }
+
+        columns = bigquery_service.tables().get(
```

Comment:
from here to the end of the function - isn't a functionality like this already exists (haven't looked in) and can be reused (and by that making the introduced changes footprint smaller )

Reply:
No, Redash already has some sort of facility to display data, however...

When redash displays data it reads this data from Redis(where the results of a query are stored), the column names are read from the cursor configuration, and the whole thing is behind 20 layers of different abstractions.

In this case the column names are the table columns, and the data is mapped into a structure 
MDBDataTable - https://mdbootstrap.com/docs/react/tables/datatables/
expects(which is a new dependency this PR introduces), which has nothing to do with how Redash displays data because of all those abstraction layers.

Reply:
The API call that is reused is the call to list table structure, but even if you extract it, you're not going to save anything since the data is formatted differently due to redash and MDBDataTable having different data structure requirements.